component/src/main/java/org/wso2/extension/siddhi/io/websocket/sink/websocketserver/WebSocketServerSink.java

```diff
@@ -0,0 +1,212 @@
+import org.wso2.extension.siddhi.io.websocket.util.WebSocketUtil;
+import org.wso2.siddhi.annotation.Example;
+import org.wso2.siddhi.annotation.Extension;
+import org.wso2.siddhi.annotation.Parameter;
+import org.wso2.siddhi.annotation.util.DataType;
+import org.wso2.siddhi.core.config.SiddhiAppContext;
+import org.wso2.siddhi.core.exception.ConnectionUnavailableException;
+import org.wso2.siddhi.core.stream.output.sink.Sink;
+import org.wso2.siddhi.core.util.config.ConfigReader;
+import org.wso2.siddhi.core.util.transport.DynamicOptions;
+import org.wso2.siddhi.core.util.transport.OptionHolder;
+import org.wso2.siddhi.query.api.definition.StreamDefinition;
+
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * {@code WebsocketServerSink } Start the WebSocket server and publishing the siddhi events.
+ */
+
+@Extension(
+        name = "websocket-server",
+        namespace = "sink",
+        description = "A Siddhi application can be configured to publish events via the WebSocket transport by " +
+                "adding the @Sink(type = ‘websocket-server’) annotation at the top of an event stream definition.",
+        parameters = {
+                @Parameter(
+                        name = "host",
+                        description = "host of the WebSocket server",
+                        type = DataType.STRING
+                ),
+                @Parameter(
+                        name = "port",
+                        description = "port of the WebSocket server",
+                        type = DataType.STRING
+                ),
+                @Parameter(
+                        name = "sub.protocol",
+                        description = "Sub-Protocols which are allowed by the service.\n" +
+                                "The sub.protocol should adhere to `subprotocol1, subprotocol2,...` format.",
+                        type = DataType.STRING,
+                        optional = true,
+                        defaultValue = "null"
+                ),
+                @Parameter(
+                        name = "idle.timeout",
+                        description = "Idle timeout of the connection. If the idle.timeout = '-1' then the timer is "
+                                + "disabled.",
+                        type = DataType.INT,
+                        optional = true,
+                        defaultValue = "-1"
+                ),
+                @Parameter(
+                        name = "tls.enabled",
+                        description = "This parameter specifies whether a secure connection is enabled or not. When " +
+                                "this parameter is set to `true`, the `keystore.path` and the `keystore.password` " +
+                                "parameters are initialized.",
+                        type = {DataType.BOOL},
+                        optional = true, defaultValue = "false"),
```

Comment:
Make it two lines? For consistency with above parameters

Reply:
fixed in 39e7381